lib/pipeline.js

```diff
@@ -191,6 +191,37 @@ Pipeline.prototype = {
   },
 
   /**
+   * Insert given streams into the pipeline at a given index
+   * @param {Number} index
+   * @param {...Stream} streams
+   * @returns {Stream[]} inserted
+   */
+  insert: function( index ) {
+
+    if( typeof index !== 'number' ) {
+      throw new Error( 'Insertion index must be a number' )
+    }
+
+    var streams = Array.prototype.slice.call( arguments, 1 )
+    var argv = [ index, 0 ].concat( streams )
+
+    this.splice.apply( this, argv )
+
+    return streams
```

Comment:
Looks like you forgot to update the `pipeline.insert` documentation?
And if you could add tests for the `insert` and `remove` return values that would be cool too :-)